Source/loadsave.cpp

```diff
@@ -363,11 +363,13 @@ void LoadPlayer(LoadHelper &file, Player &player)
 	player._pSpellFlags = file.NextLE<uint8_t>();
 	file.Skip(3); // Alignment
 
-	for (auto &spell : player._pSplHotKey)
-		spell = static_cast<spell_id>(file.NextLE<int32_t>());
-
-	for (auto &spellType : player._pSplTHotKey)
-		spellType = static_cast<spell_type>(file.NextLE<int8_t>());
+	// Extra hotkeys: to keep single player save compatibility, read only 4 hotkeys here, rely on LoadHotkeys for the rest
+	for (size_t i = 0; i < 4; i++) {
+		player._pSplHotKey[i] = static_cast<spell_id>(file.NextLE<int32_t>());
+	}
+	for (size_t i = 0; i < 4; i++) {
+		player._pSplTHotKey[i] = static_cast<spell_type>(file.NextLE<int8_t>());

```

Comment:
This set of reads (and the matching writes in `SavePlayer`) can use `uint8_t` to match the updates you made when loading the hotkeys file. uint8_t is more suitable given the cast to an enum and the values used, good to update it when you're in the area anyway :D

Reply:
The player save function writes a int8_t there, LoadHotkeys() has uint8_t, but its separate from the old style saves. I assume that the int8_t here should stay for compatibility.

The LoadHotkeys and SaveHotkeys were another thing, SaveHotkeys was writing a uint8_t, so I made the LoadHotkeys function match it because all existing saves have a uint8_t because of that.

Reply:
`spell_type` is defined with an explicit backing type of `uint8_t`, reading and writing to the main save file should really use the same type. It's not a required change because we read an 8 bit value and the expected range is 0-4 (there's no sign extension issues), however it'd be nice to have to make the relationship clearer.